src/containers/AssetForm.js

```diff
@@ -74,89 +72,75 @@ const retentionStyle = {
   padding: '10px 20px'
 };
 
+export const NEW_ASSET = {
+  name: null,
+  department: null,
+  purpose: null,
+  research: null,
+  owner: null,
+  private: null,
+  personal_data: null,
+  data_subject: [],
+  data_category: [],
+  recipients_category: null,
+  recipients_outside_eea: null,
+  retention: null,
+  risk_type: [],
+  storage_location: null,
+  storage_format: [],
+  paper_storage_security: [],
+  digital_storage_security: []
+};
+
 /*
   Renders the form for the creation/editing of an Asset.
   */
 class AssetForm extends Component {
 
   constructor() {
     super();
-
-    this.state = {
-      // asset state
-      name: null,
-      department: null,
-      purpose: null,
-      research: null,
-      owner: null,
-      private: null,
-      personal_data: null,
-      data_subject: [],
-      data_category: [],
-      recipients_category: null,
-      recipients_outside_eea: null,
-      retention: null,
-      risk_type: [],
-      storage_location: null,
-      storage_format: [],
-      paper_storage_security: [],
-      digital_storage_security: [],
-    }
+    this.state = NEW_ASSET;
   }
 
   /*
-  Wrapper for fetch() that handles errors / unmarshalls JSON
+  If the form is in edit mode - fetch the asset
    */
-  fetch(url, options, cb) {
-    if (options.headers) {
-      options.headers.append('Authorization', 'Bearer ' + ACCESS_TOKEN);
-    } else {
-      options['headers'] = new Headers({
-        'Authorization': 'Bearer ' + ACCESS_TOKEN
-      });
-    }
-    fetch(url, options).then(response => {
-      if (response.ok) {
-        return response.json()
+  componentDidMount() {
+    if (this.props.assetUrl) {
+      if (this.props.asset) {
+        this.setState(this.props.asset.asset);
       } else {
-        this.props.handleMessage('Network Error: ' + response.statusText)
+        this.props.getAsset(this.props.assetUrl);
       }
-    }).then(data => data && cb(data)).catch(
-      error => this.props.handleMessage('Network Error: ' + error)
-    );
+    }
   }
 
   /*
-  If the form is in edit mode - fetch the asset
+  Receives updated props - specifically an Asset from redux state.
    */
-  componentDidMount() {
-    if (this.props.match.url !== '/asset/create') {
-      this.fetch(config.ENDPOINT_ASSETS + this.props.match.params.assetId + '/', {}, data => {
-        this.setState(data);
-      });
-    }
+  componentWillReceiveProps(nextProps) {
+    nextProps.asset && this.setState(nextProps.asset.asset);
```

Comment:
Exploiting shortcut logical operator evaluation is essential in shell, idiomatic in C but perhaps JavaScript can make do with the less clever but more boring
```js
if(nextProps.asset) { this.setState(nextProps.asset.asset); }
```